cumulus/parachains/runtimes/assets/asset-hub-rococo/src/lib.rs

```diff
@@ -869,6 +872,38 @@ impl pallet_xcm_bridge_hub_router::Config<ToWestendXcmRouterInstance> for Runtim
 	type FeeAsset = xcm_config::bridging::XcmBridgeHubRouterFeeAssetId;
 }
 
+/// XCM router instance to BridgeHub with bridging capabilities for `Ethereum` global
```

Comment:
Is there a point of adding this pallet instance to the asset hub runtime? This pallet is an equivalent of the `SovereignPaidRemoteExporter`, which supports dynamic fee factor. TLDR: pallet has two modes: "congested" and "uncongested". We switch those modes when we are sending `vec![Transact { call: report_bridge_status(<congested-or-not>) }]` messages from Bridge Hub to Asset Hub. This is done by our messages pallet [here](https://github.com/paritytech/polkadot-sdk/blob/4c4407a8930267e139db93019c01444acf49e1ce/bridges/bin/runtime-common/src/messages_xcm_extension.rs#L203-L241). And it happens when we see that there are too many messages in the outbound bridge queue.

But I can't find any code that sends the same message from Rococo BH to this pallet instance at Rococo AH, which essentially turns the pallet into simple `SovereignPaidRemoteExporter`. So imo it is simpler to use it here, if I'm not missing anything

Reply:
Good point. We should use `SovereignPaidRemoteExporter` instead then, as our bridge doesn't currently send back-pressure signals to AssetHub.

However this raises the question of whether we actually do need to send back-pressure signals to AssetHub.

On BridgeHub:
1. Outbound messages to Ethereum are buffered in `MessageQueue`.
2. Our BridgeHub exporter is able to export 32 (or more, this is configurable) messages per block, and there is no head-of-line blocking on the Ethereum side, so once a message is exported, it is considered processed.
3. Suppose for some exceptional, emergency reason we need to halt the exporting of messages. Then messages from AssetHub will pile up in the MessageQueue. This should be safe, as long as the storage cost is accounted for in XCM fees charged by BridgeHub, right?

The whole process is documented more in [here](https://github.com/Snowfork/snowbridge/blob/c03175b813ee97373843de4f70ea10df1977beb7/parachain/pallets/outbound-queue/src/lib.rs#L5).

Note: At current Ethereum gas prices, AssetHub & BridgeHub are going to charge over $20 for every exported message.

Reply:
> Our BridgeHub exporter is able to export 32 (or more, this is configurable) messages per block, and there is no head-of-line blocking on the Ethereum side, so once a message is exported, it is considered processed.

True, just wanted to mention that's an upper bound. In practice it could process less than this number because there are also other bridges, transactions and sibling chains that are "eating" the block.

> Suppose for some exceptional, emergency reason we need to halt the exporting of messages. Then messages from AssetHub will pile up in the MessageQueue. This should be safe, as long as the storage cost is accounted for in XCM fees charged by BridgeHub, right?
> Note: At current Ethereum gas prices, AssetHub & BridgeHub are going to charge over $20 for every exported message.

IMO $20 is enough to protect us from possible spam attacks. But in general IMO you need a way to adjust fee that is paid by the **sender** at the sending chain (asset hub)  if anything is changed at bridge hub (gas price change? conversion rate change? ...). IIUC all gas-related computations happen at the bridge hub, so maybe let's allow governance to change fee at asset hub? In other words - do you think it makes sense to turn `const BridgeHubEthereumBaseFeeInROC` to `pub storage BridgeHubEthereumBaseFeeInROC` (and also add corresponding calls that may change this const to the `SafeCallFilter`).

Reply:
(not that important for Rococo<>Ethereum bridge, but since that's a prototype for Polkadot<>Ethereum bridge, imo it makes sense to do that here if you agree)

Reply:
Gotcha, ok, turning `BridgeHubEthereumBaseFeeInROC` into a storage item that can be changed via a privileged `frame_system::Call::set_storage` seems simple enough.

Agree that we should make that change, as the alternative would be a whole runtime upgrade which is quite excessive.

Will make that change 👍 

> True, just wanted to mention that's an upper bound. In practice it could process less than this number because there are also other bridges, transactions and sibling chains that are "eating" the block.

Well actually, our outbound queue pallet reserves enough block weight to process at most 32 messages, as per https://github.com/Snowfork/snowbridge/blob/c03175b813ee97373843de4f70ea10df1977beb7/parachain/pallets/outbound-queue/src/lib.rs#L239

Reply:
> Well actually, our outbound queue pallet reserves enough block weight to process at most 32 messages, as per https://github.com/Snowfork/snowbridge/blob/c03175b813ee97373843de4f70ea10df1977beb7/parachain/pallets/outbound-queue/src/lib.rs#L239

Oh, missed that. Thank you for clarification

Reply:
Actually, even though you're reserving this weight, IIUC you may still process less than `32` messages, because it is the `MessageQueue` decides when and how process its messages. So if there are queued messages from other parachains and/or bridges and the message queue itself processes messages in `from_idle` iirc. So some outbound messages may stay in the message queue for a while if bridge hub is overloaded. But I personally do not see any problem with that :)